PocketKit/Tests/PocketKitTests/AppBadgeTrackerTests.swift

```diff
@@ -0,0 +1,71 @@
+import XCTest
+@testable import PocketKit
+import Combine
+import SharedPocketKit
+
+class MockAppBadgeTracker: AppBadgeTracker {
+    var getManualCheckForSavedCountGotCalled = false
+
+    private let source: MockSource!
+    private let notificationCenter: NotificationCenter
+    private var subscriptions: Set<AnyCancellable> = []
+    private var userDefaults: UserDefaults
+
+    init(source: MockSource!, userDefaults: UserDefaults, notificationCenter: NotificationCenter = .default) {
+        self.source = source
+        self.notificationCenter = notificationCenter
+        self.userDefaults = userDefaults
+
+        setupNotificationSubscription()
+    }
+
+    private func setupNotificationSubscription() {
+        self.notificationCenter
+            .publisher(for: .listUpdated)
+            .sink { [weak self] _ in
+                self?.manualCheckForSavedCount()
+            }
+            .store(in: &subscriptions)
+    }
+
+    func manualCheckForSavedCount() {
+        let shouldToggle = userDefaults.bool(forKey: AccountViewModel.ToggleAppBadgeKey)
+        if shouldToggle {
+            getManualCheckForSavedCountGotCalled = true
+        } else {
+            getManualCheckForSavedCountGotCalled = false
+        }
+    }
+}
+
+final class AppBadgeTrackerTests: XCTestCase {
+
+    private var source: MockSource!
+    private var userDefaults: UserDefaults!
+    private var accountViewModel: AccountViewModel!
+    private var appBadgeTracker: MockAppBadgeTracker!
+
+    override func setUp() {
+        source = MockSource()
+        userDefaults = UserDefaults()
+        appBadgeTracker = MockAppBadgeTracker(source: source, userDefaults: userDefaults)
```

Comment:
curious - if we make changes to the `AppBadgeTracker` class, then would these tests still pass since we are testing the mock vs the actual class?

Reply:
yup, if AppBadgeTracker changes, we will have to update this mock class.  I was looking at the other `Mock` classes that we have to try and copy the behavior, and they all seem to be similar that they are based around a protocol.  If thats not the case and I misread the code, please let me know and I will change it (but will need help with solution)